Lib/test/test_tarfile.py

```diff
@@ -1686,7 +1940,33 @@ def test_pax_limits(self):
 
 class MiscTest(unittest.TestCase):
 
+    # Came in on Python3.6
+    def test_char_fields(self):
+        self.assertEqual(tarfile.stn("foo", 8),
+                          b"foo\0\0\0\0\0")
+        self.assertEqual(tarfile.stn("foobar", 3),
+                          b"foo")
+        self.assertEqual(tarfile.nts(b"foo\0\0\0\0\0"),
+                          "foo")
+        self.assertEqual(tarfile.nts(b"foo\0bar\0"),
+                          "foo")
+    
     def test_read_number_fields(self):
+        # Issue 13158: Test if GNU tar specific base-256 number fields
+        # are decoded correctly.
+        self.assertEqual(tarfile.nti(b"0000001\x00"), 1)
+        self.assertEqual(tarfile.nti(b"7777777\x00"), 0o7777777)
+        self.assertEqual(tarfile.nti(b"\x80\x00\x00\x00\x00\x20\x00\x00"),
+                       0o10000000)
+        self.assertEqual(tarfile.nti(b"\x80\x00\x00\x00\xff\xff\xff\xff"),
+                       0xffffffff)
+        # self.assertEqual(tarfile.nti(b"\xff\xff\xff\xff\xff\xff\xff\xff"),
```

Comment:
What's the reason for not running all these checks?

Reply:
Those are for unsupported header types